mods/taskbar-clock-customization.wh.cpp

```diff
@@ -1070,6 +1124,346 @@ void WebContentUpdateThreadUninit() {
+                        // Truncate with ellipsis
+                        if (g_settings.mediaInfoMaxLength >= 3) {
+                            g_mediaInfoEllipsisFormatted.buffer[g_settings.mediaInfoMaxLength - 3] = L'.';
+                            g_mediaInfoEllipsisFormatted.buffer[g_settings.mediaInfoMaxLength - 2] = L'.';
+                            g_mediaInfoEllipsisFormatted.buffer[g_settings.mediaInfoMaxLength - 1] = L'.';
+                            g_mediaInfoEllipsisFormatted.buffer[g_settings.mediaInfoMaxLength] = L'\0';
+                        } else {
+                            g_mediaInfoEllipsisFormatted.buffer[g_settings.mediaInfoMaxLength] = L'\0';
+                        }
+                    }
+                }
+            }
+
+            // Get playback status
+            auto status = playbackInfo.PlaybackStatus();
+            switch (status) {
+                case winrt::Windows::Media::Control::GlobalSystemMediaTransportControlsSessionPlaybackStatus::Playing:
+                    wcscpy_s(g_mediaStatusFormatted.buffer, L"Playing");
+                    break;
+                case winrt::Windows::Media::Control::GlobalSystemMediaTransportControlsSessionPlaybackStatus::Paused:
+                    wcscpy_s(g_mediaStatusFormatted.buffer, L"Paused");
+                    break;
+                case winrt::Windows::Media::Control::GlobalSystemMediaTransportControlsSessionPlaybackStatus::Stopped:
+                    wcscpy_s(g_mediaStatusFormatted.buffer, L"Stopped");
+                    break;
+                default:
+                    wcscpy_s(g_mediaStatusFormatted.buffer, L"Unknown");
+                    break;
+            }
+
+            // Hide media info if paused/stopped and hideMediaOnPause is enabled
+            if (g_settings.hideMediaOnPause && 
+                (status == winrt::Windows::Media::Control::GlobalSystemMediaTransportControlsSessionPlaybackStatus::Paused ||
+                 status == winrt::Windows::Media::Control::GlobalSystemMediaTransportControlsSessionPlaybackStatus::Stopped)) {
+                wcscpy_s(g_mediaTitleFormatted.buffer, L"");
+                wcscpy_s(g_mediaArtistFormatted.buffer, L"");
+                wcscpy_s(g_mediaAlbumFormatted.buffer, L"");
+                wcscpy_s(g_mediaInfoFormatted.buffer, L"");
+                wcscpy_s(g_mediaInfoEllipsisFormatted.buffer, L"");
+            }
+
+        } catch (...) {
+            wcscpy_s(g_mediaTitleFormatted.buffer, L"");
+            wcscpy_s(g_mediaArtistFormatted.buffer, L"");
+            wcscpy_s(g_mediaAlbumFormatted.buffer, L"");
+            wcscpy_s(g_mediaStatusFormatted.buffer, L"Error");
+            wcscpy_s(g_mediaInfoFormatted.buffer, L"");
+            wcscpy_s(g_mediaInfoEllipsisFormatted.buffer, L"");
+        }
+
+        g_mediaLastUpdateTime = expectedUpdateTime;
+    }
+}
+
+void MediaSessionInit() {
+    try {
```

Comment:
You can optimize this by only initializing `g_mediaSessionManager` if any of the relevant patterns are used, like it's done in other places. Something like:
```cpp
  if (!IsStrInDateTimePatternSettings(L"%media_...%") && ...) {
    return;
  }
```